clients/unshaded/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -58,12 +58,14 @@ public static BlockOutStream get(TachyonFile tachyonFile, WriteType opType, int
    */
   public static BlockOutStream get(TachyonFile tachyonFile, WriteType opType, int blockIndex,
       long initialBytes, TachyonConf tachyonConf) throws IOException {
-
-    if (tachyonFile.mTachyonFS.hasLocalWorker()
-        && tachyonConf.getBoolean(Constants.USER_ENABLE_LOCAL_WRITE,
-            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)) {
+    if ((tachyonFile.mTachyonFS.hasLocalWorker()
+            && tachyonConf.getBoolean(Constants.USER_ENABLE_LOCAL_WRITE,
+            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)
+            && tachyonConf.get(
```

Comment:
What happens if you don't add this condition? I'm not sure I understand why the added condition is required.